### PR TITLE
[COMP-1282] Filter Transfer 0 events emitted when miners are paid

### PR DESCRIPTION
**Pull request by waynenilsen**
These are just cluttering up the subscan with noise (and will do the same for other tools)

### CODE REVIEW

**Comment by waynenilsen** [comment]
substrate coverage failure above looks like a flaky test - it runs on my machine and the line is the assertion below

```
            // Third round should process
            let pending_blocks = PendingChainBlocks::get(ChainId::Eth);
            assert_eq!(pending_blocks.len(), 0);
```

for context

**Comment by jflatow** [comment]
> substrate coverage failure above looks like a flaky test - it runs on my machine and the line is the assertion below
> 
> ```
>             // Third round should process
>             let pending_blocks = PendingChainBlocks::get(ChainId::Eth);
>             assert_eq!(pending_blocks.len(), 0);
> ```
> 
> for context

AFAIK this test shouldn't be flaky

File: pallets/cash/src/internal/initialize.rs [hunk 1]
```diff
@@ -115,12 +115,17 @@ pub fn initialize_block<T: Config>(now: Timestamp) -> Result<(), Reason> {
...
+        // These are just cluttering up tools with noise
+        <Module<T>>::deposit_event(Event::TransferCash(
+            GATEWAY_VOID,
+            last_miner,
+            last_miner_share_principal,
+            cash_index_new,
+        ));
+    }
+
     <Module<T>>::deposit_event(Event::MinerPaid(last_miner, last_miner_share_principal));
```
Comment (jflatow) [inline pallets/cash/src/internal/initialize.rs:128]: Should we filter this one too?

File: pallets/cash/src/internal/initialize.rs [hunk 2]
```diff
@@ -115,12 +115,17 @@ pub fn initialize_block<T: Config>(now: Timestamp) -> Result<(), Reason> {
...
     }
 
-    <Module<T>>::deposit_event(Event::TransferCash(
-        GATEWAY_VOID,
-        last_miner,
-        last_miner_share_principal,
-        cash_index_new,
-    ));
+    if last_miner_share_principal != CashPrincipalAmount::ZERO {
+        // Filter Transfer 0 events emitted when miners are paid
```
Comment (jflatow) [inline pallets/cash/src/internal/initialize.rs]: Maybe just "No need to emit events when nothing happens", these wont be cluttering anything up once this is merged

File: pallets/cash/src/internal/initialize.rs (hunk 1 above)
Comment (waynenilsen) [reply pallets/cash/src/internal/initialize.rs:128]: Yeah i wasn't sure on this one that is not what was described in the ticket. I think it probably makes sense to have one of these per block, something like our version of a coinbase tx event

**Comment by waynenilsen** [comment]
hmm now `substrate coverage` went ✅ but `ethereum tests` went from ✅  to ❌ 

```
FAIL tests/cash_token_test.js (8.269s)
  ● CashToken › Upgradeable › should be upgradeable to new logic (without call)

    expected (0x6f564ea7efdd90efad8923c6a706428e2cc1e3) == (0x006F564eA7eFDd90eFAD8923C6a706428E2cC1e3)

      87 |       await proxyAdmin.methods.upgrade(cash._address, cashImpl2._address).send({ from: root });
      88 | 
    > 89 |       expect(await getProxyImplementation(cash)).toMatchAddress(cashImpl2._address);
         |                                                  ^
      90 |       expect(await getProxyAdmin(cash)).toMatchAddress(proxyAdmin._address);
      91 | 
      92 |       expect(await cash.methods.admin().call()).toEqual(account3);

      at Object.<anonymous> (tests/cash_token_test.js:89:50)
          at runMicrotasks (<anonymous>)
```

🤷 will have to look into this more .....

**Review by jflatow — APPROVED** [review]

**Pull request merged**